# Single-cell-tools

An always expanding list of software packages for single-cell analysis

## Software in R
- [cellassign](https://github.com/irrationone/cellassign/) - Automatically assigns single-cell RNA-seq data to known cell types across thousands of cells accounting for patient and batch specific effects.
- [celda](https://bioconductor.org/packages/release/bioc/html/celda.html) - A suite of Bayesian hierarchical models for clustering single-cell RNA-sequencing (scRNA-seq) data.
- [cellity](https://github.com/teichlab/cellity) - Contains functions to help to identify low quality cells in scRNA-seq data.
- [copykat](https://github.com/navinlabcode/copykat) - Inference of genomic copy number and subclonal structure of human tumors from high-throughput single cell RNAseq data.
- [DALI](https://github.com/vibscc/DALI) - R-package for the analysis of single-cell TCR/BCR data in the Seurat ecosystem.
- [dittoSeq](https://github.com/dtm2451/dittoSeq) - A set of functions built to enable analysis and visualization of single-cell and bulk RNA-sequencing data by novice, experienced, and color blind coders.
- [GARNET](https://cole-trapnell-lab.github.io/garnett/) - Automated cell type classification.
- [inferCNV](https://github.com/broadinstitute/inferCNV) - Inferring copy number alterations from tumor single cell RNA-Seq data.
- [Monocle](https://cole-trapnell-lab.github.io/monocle-release/) - An analysis toolkit for single-cell RNA-seq.
- [Muscat](https://github.com/HelenaLC/muscat) - Various methods for Differential State (DS) analyses in multi-sample, multi-group, multi-(cell-)subpopulation scRNA-seq data.
- [numbat](https://github.com/kharchenkolab/numbat) - A haplotype-aware CNV caller from single-cell and spatial transcriptomics data.
- [SCEVAN](https://github.com/AntonioDeFalco/SCEVAN) - A variational algorithm to detect the clonal copy number substructure of tumors from scRNA-seq data.
- [SCP](https://github.com/zhanghao-njmu/SCP) - Provides a comprehensive set of tools for single-cell data processing and downstream analysis.
- [Seurat](https://satijalab.org/seurat/) - R toolkit for single cell genomics.
- [slingshot](https://github.com/kstreet13/slingshot) - Designed to model developmental trajectories in single-cell RNA sequencing data.  


## Software in Python
- [infercnvpy](https://github.com/icbi-lab/infercnvpy) - Scanpy plugin to infer copy number variation (CNV) from single-cell transcriptomics data.
- [GSEApy](https://github.com/zqfang/GSEApy) - A Python/Rust implementation for GSEA and wrapper for Enrichr.
- [kb-python](https://github.com/pachterlab/kb_python) - A python package for processing single-cell RNA-sequencing. It wraps the kallisto|bustools single-cell RNA-seq command line tools in order to unify multiple processing workflows.
- [pySCENIC](https://github.com/aertslab/pySCENIC) - A lightning-fast python implementation of the SCENIC pipeline (Single-Cell rEgulatory Network Inference and Clustering) which enables biologists to infer transcription factors, gene regulatory networks and cell types from single-cell RNA-seq data.
- [Scanpy](https://github.com/theislab/scanpy)  - A scalable toolkit for analyzing single-cell gene expression data built jointly with anndata.
- [scbean](https://github.com/jhu99/scbean) - Integrates a range of models for single-cell data analysis, including dimensionality reduction, remvoing batch effects, and transferring well-annotated cell type labels from scRNA-seq to scATAC-seq and spatial resoved transcriptomics.
- [scirpy](https://github.com/scverse/scirpy) - A package to analyse T cell receptor (TCR) or B cell receptor (BCR) repertoires from single-cell RNA sequencing (scRNA-seq) data in Python.
- [scTCRseq](https://github.com/ElementoLab/scTCRseq)  - an implementation of a pipeline for Single-cell RNAseq package for recovering TCR data in python.
- [scVelo](https://github.com/theislab/scvelo) - A scalable toolkit for RNA velocity analysis in single cells.


  
## Software in other language
- [SCell](https://github.com/diazlab/SCell) - An integrated software tool for quality filtering, normalization, feature selection, iterative dimensionality reduction, clustering and the estimation of gene-expression gradients from large ensembles of single-cell RNA-seq datasets.
- [Falco](https://github.com/VCCRI/Falco) - A quick and flexible single-cell RNA-seq processing framework on the cloud.
- [kallisto](https://github.com/pachterlab/kallisto) - A program for quantifying abundances of transcripts from RNA-Seq data, or more generally of target sequences using high-throughput sequencing reads.
